server/pulp/server/controllers/repository.py

```diff
@@ -1006,16 +1006,17 @@ def check_publish(repo_obj, dist_id, dist_inst, transfer_repo, conduit, call_con
                                                            updated__gte=the_timestamp).count()
         units_removed = last_unit_removed is not None and last_unit_removed > last_published
         dist_updated = dist.last_updated > last_published
-        published_after_predistributor = True
     else:
         published_after_predistributor = False
     predistributor_id = call_config.get('predistributor_id')
-    if predistributor_id and last_published:
+    if predistributor_id:
         predistributor = model.Distributor.objects.get_or_404(repo_id=repo_obj.repo_id,
                                                               distributor_id=predistributor_id)
         predistributor_last_published = predistributor["last_publish"]
-        if predistributor_last_published:
+        if predistributor_last_published and last_published:
             published_after_predistributor = last_published > predistributor_last_published
+        else:
+            published_after_predistributor = True
```

Comment:
i am still trying to get the general logic here, but i think this should be set to false